platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/WizAssetsPlugin.java

```diff
@@ -67,24 +70,37 @@
     private static final int FILE_CREATION_ERROR = 7;
     private static final int JSON_CREATION_ERROR = 8;
 
-    private String pathToStorage;
+    private String pathToDatabase;
+    private String pathToAssets;
     private int blockSize;
 
     @Override
     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
         super.initialize(cordova, webView);
 
         final Context applicationContext = cordova.getActivity().getApplicationContext();
-        pathToStorage = applicationContext.getCacheDir().getAbsolutePath() + File.separator;
+
+        pathToDatabase = applicationContext.getCacheDir().getAbsolutePath() + File.separator + PLUGIN_FOLDER;
+        pathToAssets = pathToDatabase + File.separator + ASSETS_FOLDER;
+
+        if (!createFolderIfRequired(pathToDatabase)) {
+            Log.e(TAG, "error -- unable to create folder: " + pathToDatabase);
```

Comment:
Just pushed a new version including an `initialize` method with a callback informing javascript if everything is fine or if the plugin did not initialize and should not be used.